edgraph/server.go

```diff
@@ -570,6 +575,10 @@ func (s *Server) doMutate(ctx context.Context, qc *queryContext, resp *api.Respo
 		return err
 	}
 
+	if err := verifyUniqueWithinMutation(qc); err != nil {
```

Comment:
Can we maybe update the function name? or Add a comment up here to as to show what we are trying to do? 
What is the behaviour today if the user gave two different mutations for the same predicate on the same uid? Have we handled that here? Is user giving two mutations, for same uid, for same predicate allowed for unique today? 
What I am thinking is this could be a in general function that validates duplicates in the same predicate. Irrespective of unique or not. For unique we could have one more check.

Reply:
We are literally trying to check unique within the mutation. What did you feel missing from the function name?

Why would we want to check for duplicate mutation if the predicate is not marked unique?